bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsBinarySyncImpl.java

```diff
@@ -0,0 +1,209 @@
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+import javax.jcr.Value;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Service to sync a remote asset's binaries a from remote server. Implements {@link RemoteAssetsBinarySync}.
+ */
+@Component(
+    label = "ACS AEM Commons - Remote Assets - Binary Sync Service",
+    description = "Pulls the binaries for a remote asset in order to make it a true local asset."
+)
+@Service
+public class RemoteAssetsBinarySyncImpl implements RemoteAssetsBinarySync {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RemoteAssetsBinarySyncImpl.class);
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Reference
+    private RemoteAssetsConfig remoteAssetsConfig;
+
+    private ResourceResolver resourceResolver;
+    private Session session;
+
+    /**
+     * Method to run on activation.
+     * @throws RepositoryException exception
+     */
+    @Activate
+    protected void activate() throws RepositoryException {
+        this.resourceResolver = RemoteAssets.logIn(this.resourceResolverFactory);
+        this.session = this.resourceResolver.adaptTo(Session.class);
+        if (StringUtils.isNotBlank(this.remoteAssetsConfig.getEventUserData())) {
+            this.session.getWorkspace().getObservationManager().setUserData(this.remoteAssetsConfig.getEventUserData());
+        }
+    }
+
+    /**
+     * Method to run on deactivation.
+     */
+    @Deactivate
+    protected void deactivate() {
+        if (this.session != null) {
+            try {
+                this.session.logout();
+            } catch (Exception e) {
+                LOG.warn("Failed session.logout()", e);
+            }
+        }
+        if (this.resourceResolver != null) {
+            try {
+                this.resourceResolver.close();
+            } catch (Exception e) {
+                LOG.warn("Failed resourceResolver.close()", e);
+            }
+        }
+    }
+
+    /**
+     * @see RemoteAssetsBinarySync#syncAsset(Resource)
+     * @param resource Resource
+     * @return Resource
+     */
+    @Override
+    public Resource syncAsset(Resource resource) {
+        try {
+            this.session.refresh(true);
+            Resource localRes = this.resourceResolver.getResource(resource.getPath());
+            Node node = localRes.adaptTo(Node.class);
+
+            Asset asset = localRes.getParent().adaptTo(Asset.class);
```

Comment:
Is it better to use DamUtil.resolveToAsset(..)  since this is a public API (not private).

does it need nullchecks?

Reply:
Didn't realize DamUtil existed.  I'm fine with using it here if you think that we should.

A null check probably isnt a terrible idea as well - it *should* never happen since this service should only be called on a dam:AssetContent node, but it doesnt hurt to handle the case if someone were to pass something crazy in or if theres a bug somewhere else.